tower/src/util/call_all/common.rs

```diff
@@ -111,6 +111,7 @@ where
                 Poll::Pending => {
                     // TODO: We probably want to "release" the slot we reserved in Svc here.
                     // It may be a while until we get around to actually using it.
+                    return Poll::Pending;
```

Comment:
nit, take it or leave it: now that we return `Poll::Pending` in this case, we could simplify the nested `match` to just a single `match` by using `ready!` instead:
```rust

            // If it is, gather the next request (if there is one), or return `Pending` if the
            // stream is not ready.
            // TODO: We probably want to "release" the slot we reserved in Svc if the
            // stream returns `Pending`. It may be a while until we get around to actually
            // using it.
            match ready!(this.stream.as_mut().poll_next(cx)) {
                Some(req) => {
                    this.queue.push(svc.call(req));
                }
                None => {
                    // We're all done once any outstanding requests have completed
                    *this.eof = true;
                }
            }
 ```

Reply:
Thank you for the review, I've done this refactor.